pkg/server/server.go

```diff
@@ -653,15 +653,26 @@ func (s *Server) ListUsers(
 	ctx context.Context,
 	req *openfgav1.ListUsersRequest,
 ) (*openfgav1.ListUsersResponse, error) {
+
 	typesys, err := s.typesystemResolver(ctx, req.GetStoreId(), req.GetAuthorizationModelId())
 	if err != nil {
 		return nil, err
 	}
 
+	err = listusers.ValidateListUsersRequest(req, typesys)
+	if err != nil {
+		return nil, err
+	}
+
 	ctx = typesystem.ContextWithTypesystem(ctx, typesys)
 
+	for _, ctxTuple := range req.GetContextualTuples().GetTupleKeys() {
+		if err := validation.ValidateTuple(typesys, ctxTuple); err != nil {
+			return nil, serverErrors.HandleTupleValidateError(err)
+		}
+	}
```

Comment:
Suggestion (not-blocking): could we experiment with putting this code inside `internal/server/list_users.go`? Then we can move the validation unit tests to `internal/server/list_users_test.go`. I think `server_test.go` has gotten too big and it mixes tests that are very unrelated.

Reply:
Suggestion to the suggestion: Start introducing per RPC test files so that we don't have a massive `server_test.go` file.

e.g.

```
pkg/
  server/
    server.go
    server_test.go
    check_test.go
    list_objects_test.go
    ...
```